Initial cut for repository to hold all migrations

asarfara opened the pull request:
This feature introduces migration of events from content vic to any other website. It uses drupal migrate modules and json api to fetch content from content vic and parse it to a default event content type.

**Added:**
- Mapping of event data to be used by migrate module
- Unit testing of the business logic defined in services and mapper.

**Remaining:**
- Add uninstall hook to remove any migration added by this module.
- Use it on a project to check if any changes could have broken the migration.

**Referenced Ticket**
https://digital-engagement.atlassian.net/browse/TVD-17

asarfara committed:
Added event migration components.

A new module has been created i.e. tide_migration which will serve as a central repository to holding all sorts of migration that will take content from content.vic to any other website.

asarfara committed:
Added test to cover events mapper.

Addition of unit tests will ensure our business logic conditions are covered.

asarfara committed:
Introduced phpunit testing to tide_migration.

Added drupal unit testing classes to make sure we can test drupal based methods and objects.

anthony-malkoun commented:
Hey @asarfara this looks like an amazing start!

Can I ask for a couple of things.
1. Can we have some instructions on how to run the tests that you've created (an ahoy command might be a good space for this)
2. Ideally these tests should be running on push to a branch in CircleCI. I can help you with that next week if you'd like.
3. Maybe make this a draft PR for now while we're working on it.

asarfara commented:
Hi @anthony-malkoun 
Yep i would need your help to get this module running on circle ci. Good one about adding a command to ahoy, I will do that :)

asarfara committed:
Added unit test command to ahoy file so that we can just type ahoy test-unit to run all tests

asarfara committed:
Moved setting file to install location so the project can override it

asarfara committed:
Renamed migration file to match module name

asarfara committed:
Add uninstall hook to remove all config install by the module

asarfara committed:
Fixed typo in the field name

asarfara committed:
Updated database condition to remove only files that matches this module

asarfara committed:
Removed file handling and parsing yml. Instead of that we can just use drupal config value get to fetch overridden values

asarfara committed:
Instead of fetching a file and parsing yaml content, we can just use drupal config get to fetch the overidden value we require

asarfara committed:
Renamed all files to be in install folder instead of optional folder

asarfara committed:
Removed condition where the variable could be a string or an integer

asarfara committed:
Corrected index when passing details to generate paragraph

asarfara committed:
Added new object that can be used by other process plugins instead of loading a whole process plugin when only one function was required

asarfara committed:
Added process to skip when a taxonomy already exists in the system. This will prevent duplication of the taxonomy terms

asarfara committed:
Removed config fetcher as it can be more simplified by just injecting the correct component

asarfara committed:
Fixed spacing between statements

asarfara commented:
Hey @GROwen this PR is up for review now :)

asarfara committed:
Updated reserved variables to include event urls that can be configured per project.

This is to demonstrate an example so that we are not bounded to only use content.vic based urls. We can pass in any url that returns json response and is in the same format as the mapper.

asarfara committed:
Upgraded tide_core version

Commit:
updating version to D9

yeniatencio committed:
Update tide_migration.info.yml

yeniatencio committed:
Merge pull request #3 from dpc-sdp/feature/SDPA-5887-drupal-9x

[SDPA-5887]updating tide_migration version to D9

Commit:
update tide_core and core_dev versions

yeniatencio committed:
Merge pull request #4 from dpc-sdp/feature/SRM-215-update-tide-core-version

[SRW-215]update tide_core and core_dev versions

henrytranvan committed:
SDPAP-6079: Change scrape and migration functionality so that they no longer need a deployment.

henrytranvan committed:
SDPAP-6079: Update composer.json file..

henrytranvan committed:
SDPAP-6079: Update file extentions validate.

henrytranvan committed:
SDPAP-6079: Add tide migration group.

henrytranvan committed:
SDPAP-6079: Add tide migration group, remove unuse property.

henrytranvan committed:
SDPAP-6079: Add tide migration group configuration.

henrytranvan committed:
SDPAP-6079: Add tide migration group configuration.

vincent-gao committed:
Merge pull request #7 from dpc-sdp/feature/SDPAP-6079-update-tide-migration

Feature/sdpap 6079 update tide migration
